runtime/interface.cpp

```diff
@@ -586,9 +592,11 @@ void f$flush() {
 
 void f$fastcgi_finish_request(int64_t exit_code) {
   // Run custom headers handler before body processing
-  if (headers_custom_handler_function && !headers_sent && query_type == QUERY_TYPE_HTTP) {
+  if (!headers_sent && query_type == QUERY_TYPE_HTTP) {
+    if (headers_custom_handler_function) {
+      headers_custom_handler_function();
+    }
     headers_sent = true;
```

Comment:
Btw, I moved `headers_sent = true` below `headers_custom_handler_function()` because headers_sent() within a callback should be `false`